File: KisaragiTech.Dape/Base/Random/SecureRNG.cs
using System;

namespace KisaragiTech.Dape.Base.Random;

public static class SecureRNG
{
    public static byte[] GenerateBytes(int length)
    {
        using var rng = System.Security.Cryptography.RandomNumberGenerator.Create();
        var ret = new byte[length];
        rng.GetBytes(ret);
        return ret;
    }

    public static void FillBytes(Span<byte> destination)
    {
        using var rng = System.Security.Cryptography.RandomNumberGenerator.Create();
        rng.GetBytes(destination);
    }
}

File: src/main/scala/io/github/ksrgtech/dape/base/random/SecureRNG.scala
package io.github.ksrgtech.dape.base.random

import cats.effect.kernel.Sync
import java.security.SecureRandom

object SecureRNG {

  def generateBytes[F[_]: Sync](length: Int): F[Array[Byte]] =
    Sync[F].delay {
      val rng = new SecureRandom()
      val ret = new Array[Byte](length)
      rng.nextBytes(ret)
      ret
    }

  def fillBytes[F[_]: Sync](destination: Array[Byte]): F[Unit] =
    Sync[F].delay {
      val rng = new SecureRandom()
      rng.nextBytes(destination)
    }
}
